src/bp-activity/bp-activity-notifications.php

```diff
@@ -293,6 +293,42 @@ function bp_activity_remove_screen_notifications_single_activity_permalink( $act
 }
 add_action( 'bp_activity_screen_single_activity_permalink', 'bp_activity_remove_screen_notifications_single_activity_permalink' );
 
+/**
+ * Mark notification read when user visit sing post page permalink
+ *
+ * @since BuddyPress 2.0.0
+ */
+function bp_activity_remove_screen_notifications_sing_post() {
```

Comment:
@janakkaneriya007 Function name should be `bp_activity_mark_blog_post_notification_read`